### DISCUSSION TITLE
LeanCreator as editor by default

### DISCUSSION THREAD

**Discussion by stefanos82** [rochus-keller/LeanCreator#8] [General]
An interesting approach would be to use LeanCreator as an editor by default and enable the features you want upon need to turn it to a full IDE.

An example would be Sublime Text editor or VS Code that can be used to do simple things and eventually you add plugins to extend its behavior based on a language you want to use extensively.

For instance, I would like to go in a directory of mine that I have a project in C and run `LeanCreator .` and have it loaded immediately in editor, like the 2 aforementioned editors do by default.

Is it possible to achieve something like this with LeanCreator?

**Comment by rochus-keller** [comment]
LeanCreator essentially is a variant of Qt Creator which again is a full fledged IDE. You can use it as an editor if you e.g. just open a single C++ file with it; even code indexing works to some degree. 

Changing the architecture of LeanCreator is not trivial; I still spend most of my development time to find out how this complex machinery works. It's always a trade-off whether it makes sense to add a feature to LeanCreator, or whether this shouldn't better be done in the original Qt Creator (which is open source as well). Plugins are such a topic; I actually spent a lot of time to get rid of plugins and make a single file executable; so the intention of dynamically adding or removing functionality at runtime (i.e. plugins) goes diametrally in a different direction than I've taken.

**Reply by stefanos82** [reply]
Fair points, understood :+1:
